File: lab1 class/untitled/src/DecorativeCosmetics.java
public abstract class DecorativeCosmetics extends Product{
    protected final int shade;
    protected final String texture;


    protected DecorativeCosmetics(String name, String brand,  double price, int shade, String scent, String texture) {
        super(name, brand, price, scent);
        this.shade = shade;
        this.texture = texture;
    }
    public void remove(){
        System.out.println("Removing " + name + " ...");
    };

}

File: lab2_fabrica/fabrica/DecorativeCosmetics.cs
using System;
using System.Runtime.Serialization;

namespace fabrica
{
  
    [DataContract]
    [KnownType(typeof(Mascara))]
    [KnownType(typeof(Foundation))]
    public abstract class DecorativeCosmetics : Product
    {

        [DataMember]
        protected String shade;

        [DataMember] 
        protected String texture;

        public String Shade
        {
            get { return shade; }
            set { shade = value; }
        }
        public String Texture
        {
            get => texture;    
            set { texture = value; }
        }
        protected DecorativeCosmetics(String name, String brand, double price, String shade, String texture) : base( name,  brand, price)
        {
            this.shade = shade;
            this.texture = texture;
        }

    }
}
